cms/static/js/views/move_xblock_list.js

```diff
@@ -0,0 +1,170 @@
+            this.renderXBlockInfo();
+        },
+
+        render: function() {
+            HtmlUtils.setHtml(
+                this.$el,
+                this.template(
+                    {
+                        xblocks: this.childrenInfo.children,
+                        noChildText: this.getNoChildText(),
+                        categoryText: this.getCategoryText(),
+                        XBlocksCategory: this.childrenInfo.category,
+                        forwardButtonSRText: this.getForwardButtonSRText(),
+                        currentLocationIndex: this.getCurrentLocationIndex()
+                    }
+                )
+            );
+            Backbone.trigger('move:childrenRendered', this.breadcrumbInfo());
+            return this;
+        },
+
+        renderChildren: function(event) {
+            this.renderXBlockInfo(
+                'forward',
+                $(event.target).closest('.xblock-item').data('itemIndex')
+            );
+        },
+
+        handleBreadcrumbButtonPress: function(newParentIndex) {
+            this.renderXBlockInfo('backward', newParentIndex);
+        },
+
+        renderXBlockInfo: function(direction, newParentIndex) {
+            if (direction === undefined) {
+                this.parentInfo.parent = this.model;
+            } else if (direction === 'forward') {
+                // clicked child is the new parent
+                this.parentInfo.parent = this.childrenInfo.children[newParentIndex];
+            } else if (direction === 'backward') {
+                // new parent will be one of visitedAncestors
+                this.parentInfo.parent = this.visitedAncestors[newParentIndex];
+                // remove visited ancestors
+                this.visitedAncestors.splice(newParentIndex);
+            }
+
+            this.visitedAncestors.push(this.parentInfo.parent);
+
+            if (this.parentInfo.parent.get('child_info')) {
+                this.childrenInfo.children = this.parentInfo.parent.get('child_info').children;
+            } else {
+                this.childrenInfo.children = [];
+            }
+
+            this.setDisplayedXBlocksCategories();
+            this.render();
+        },
+
+        setDisplayedXBlocksCategories: function() {
+            this.parentInfo.category = XBlockUtils.getXBlockType(
+                this.parentInfo.parent.get('category'),
+                // TODO! improve `this.visitedAncestors.length - 2` mysterious calculation
+                this.visitedAncestors[this.visitedAncestors.length - 2]
+            );
+            this.childrenInfo.category = this.categoryRelationMap[this.parentInfo.category];
+        },
+
+        getCurrentLocationIndex: function() {
+            var category, ancestorXBlock, currentLocationIndex;
+
+            if (this.childrenInfo.category === 'component' || this.childrenInfo.children.length === 0) {
```

Comment:
Don't we want to show `Current Location` on the actual (source) component being clicked ?